FLT/Deformations/Subfunctor.lean

```diff
@@ -0,0 +1,213 @@
+-- **Modified from `Subpresheaf.lean` in mathlib**
+import Mathlib
+
+/-!
+
+# Subfunctor of types
+
+We define the Subfunctor of a type valued functor.
+
+## Main results
+
+- `CategoryTheory.Subfunctor` :
+  A Subfunctor of a functor of types.
+
+-/
+
+
+universe w v u
+
+open Opposite CategoryTheory
+
+namespace CategoryTheory
+
+variable {C : Type u} [Category.{v} C]
+
+/-- A Subfunctor of a functor consists of a subset of `F.obj U` for every `U`,
+compatible with the restriction maps `F.map i`. -/
+@[ext]
+structure Subfunctor (F : C ⥤ Type w) where
+  /-- If `G` is a sub-functor of `F`, then the sections of `G` on `U` forms a subset of sections of
+    `F` on `U`. -/
+  obj : ∀ U, Set (F.obj U)
```

Comment:
I'm not sure why there isn't a SetLike instance on Subpresheaf; the PartialOrder instance would be automatic if there is. And we can probably construct a Galois insertion to `Set (Π U, Set (F.obj U))`.